File: applicatie/src/main/java/com/vdzon/administratie/checkandfix/FixService.java
package com.vdzon.administratie.checkandfix;

import com.vdzon.administratie.checkandfix.actions.fix.BoekingenFix;
import com.vdzon.administratie.checkandfix.model.CheckAndFixRegel;
import com.vdzon.administratie.model.Administratie;

import javax.inject.Inject;
import java.util.ArrayList;
import java.util.List;

public class FixService {

    private CheckService checkService;
    private List<FixFunction> fixFunctions;

    @Inject
    public FixService(CheckService checkService) {
        fixFunctions = new ArrayList<>();
        fixFunctions.add(BoekingenFix::fixAfschrift);
        this.checkService = checkService;
    }

    public Administratie getFixedAdministratie(Administratie administratie) throws Exception {
        List<CheckAndFixRegel> regelsToFix = checkService.getCheckAndFixRegels(administratie);
        final Wrapper wrapper = new Wrapper();
        wrapper.adm = administratie;
        regelsToFix.stream()
                .forEach(regel -> fixFunctions.stream().forEach(f -> wrapper.adm = f.fixAfschrift(regel, wrapper.adm))
                );
        return wrapper.adm;
    }

    class Wrapper {
        public Administratie adm = null;
    }

}

File: checkandfix/src/main/java/com/vdzon/administratie/checkandfix/FixService.kt
package com.vdzon.administratie.checkandfix

import com.vdzon.administratie.checkandfix.actions.fix.BoekingenFix2
import com.vdzon.administratie.checkandfix.model.CheckAndFixRegel2
import com.vdzon.administratie.model.Administratie
import java.util.*

/**
 * Created by robbe on 2/12/2017.
 */
object FixService {

    val fixAfschriftCheck = { regel: CheckAndFixRegel2, administratie: Administratie -> BoekingenFix2.removeBoekingen(regel,administratie)}
    val fixFunctions = Arrays.asList(fixAfschriftCheck)

    fun getFixedAdministratie(administratie: Administratie): Administratie {
        val regelsToFix = CheckService.getCheckAndFixRegels(administratie)
        var newAdministratie = administratie
        regelsToFix.forEach{regel -> fixFunctions.forEach{f -> newAdministratie = f(regel, newAdministratie)}}
        return newAdministratie
    }
}
